### DISCUSSION TITLE
Module structure - to `mod.rs` or to not `mod.rs`? that is the question.

### DISCUSSION THREAD

**Discussion by Ben-PH** [massalabs/massa#3459] [Code Style]
This question emerged from #3441. Do we go for a structure that looks like this:

```
├── foo
│   ├── mod.rs
│   └── foo_sub_module.rs
├── ledger_db.rs
├── ledger.rs
├── lib.rs
...
```
or more like this?

```
.
├── foo.rs
├── foo
│   └── foo_sub_module.rs
├── ledger_db.rs
├── ledger.rs
├── lib.rs
...
```
or is it one of those things that spending any time on it is just time lost to bike-shedding?

For me, I prefer not to have `mod.rs` in my directory structure for various reasons. Without going into detail, I just find things are easier to work with in that case.

**Comment by AurelienFT** [comment]
We often use the `mod.rs` in test to specify to export or not some of the public functions/methods depending on the features we are compiling with (often `testing` or not)

**Reply by Ben-PH** [reply]
We could have a guideline that is something like "as a rule of thumb, for non-testing modules use `module_name.rs` instead of `module_name/mod.rs`", with sub-modules added to a `module_name` directory.

My personal, opinionated stance, is that it should also apply to test modules, but that opinion doesn't take into account the context of our project. If we end up trying, and really liking, the "avoid `mod.rs` with non-test modules", we can revisit the idea of applying it to test modules as well.

**Reply by AurelienFT** [reply]
Do you have current example when we have mod.rs in sub-modules ?

**Reply by Ben-PH** [reply]
https://github.com/massalabs/massa/tree/main/massa-models/src/config

**Reply by AurelienFT** [reply]
In your mind we should have multiple files directly at the root of module or one big file ?

**Reply by Ben-PH** [reply]
In my mind, `mv mod.rs ../config.rs` would be in line with what I'm proposing.

**Reply by AurelienFT** [reply]
Does that work to have import of submodule in a file of the parent directory ?
